frontend/packages/kubevirt-plugin/src/k8s/requests/vm/actions.ts

```diff
@@ -48,26 +44,21 @@ export const VMActionWithPXERequest = async (vm: VMKind, action: VMActionType) =
 export const startVM = async (vm: VMKind) => VMActionWithPXERequest(vm, VMActionType.Start);
 export const stopVM = async (vm: VMKind) => VMActionRequest(vm, VMActionType.Stop);
 export const restartVM = async (vm: VMKind) => VMActionWithPXERequest(vm, VMActionType.Restart);
-export const deleteVM = async (vm: VMKind) => {
+export const deleteVM = (vm: VMKind) => {
   const vmImportOwnerReference = (getOwnerReferences(vm) || []).find((reference) =>
     compareOwnerReference(reference, buildOwnerReferenceForModel(VirtualMachineImportModel), true),
   );
 
+  // Note(yzamir): to delete the VM, we will delete the owner VirtualMachineImportModel.
   if (vmImportOwnerReference) {
-    const namespace = getNamespace(vm);
-    await k8sKill(VirtualMachineImportModel, {
-      metadata: { name: vmImportOwnerReference.name, namespace },
+    deleteModal({
```

Comment:
this one deletes VirtualMachineImportModel and then it tries to delete the VM in case it lost the reference meanwhile (import completed).

Also there shouldn't be any UI stuff in this file as it is under `k8s/requests`

Can you please rather just create simple confirmationModal for now? I will followup on the more advanced flow later.

Reply:
>  there shouldn't be any UI stuff in this file as it is under k8s/requests

right ! :+1: will move it

> Can you please rather just create simple confirmationModal for now?

not sure we want to create a new modal for bug fixing ... prefer to use existing modal ... 
will test and see if we can make one with very small footprint

Reply:
no need for a new one, please see confirmModal in menu-actions.ts

Reply:
nice ! using `confirmModal` now :balloon: